packages/ingest/queue.js

```diff
@@ -12,70 +11,72 @@ const {
 } = require('@cumulus/common/aws');
 
 /**
-* Create a message from a template stored on S3
-* @param {object} event
-* @param {object} event.config
-* @param {array} event.config.templates
-* @param {object} event.config.cumulus_meta
-* @param {object} event.config.cumulus_meta.config
-* @param {string} event.config.cumulus_meta.config.next name of the next function in the workflow
-* @returns {object} message object
-**/
-async function getTemplate(event) {
-  const config = event.config;
-  const templates = get(config, 'templates');
-  const nextTask = get(config, 'cumulus_meta.config.next', 'ParsePdr');
-
-  const parsedS3Uri = parseS3Uri(templates[nextTask]);
+  * Create a message from a template stored on S3
+  *
+  * @param {string} templateUri - S3 uri to the workflow template
+  * @param {Object} provider - Cumulus provider object
+  * @param {Object} collection - Cumulus collection object
+  * @returns {Promise} message object
+  **/
+async function getTemplate(templateUri, provider, collection) {
+
+  const parsedS3Uri = parseS3Uri(templateUri);
   const data = await getS3Object(parsedS3Uri.Bucket, parsedS3Uri.Key);
   const message = JSON.parse(data.Body);
 
-  message.provider = config.provider;
-  message.collection = config.collection;
-  message.meta = config.meta;
+  message.meta.provider = provider;
+  message.meta.collection = collection;
 
   return message;
 }
 
 /**
-* Create a message from a template stored on S3
-* @param {object} event
-* @param {object} event.config
-* @param {object} event.config.queues
-* @param {string} event.config.queues.startSF
-* @param {object} pdr
-* @param {string} pdr.name
-* @returns {promise} promise returned from SQS.sendMessage()
-**/
-async function queuePdr(event, pdr) {
-  const queueUrl = event.config.queues.startSF;
-  const message = await getTemplate(event);
-
-  message.input = { pdr };
+  * Create a message from a template stored on S3
+  *
+  * @param {string} queueUrl - The SQS url
+  * @param {string} templateUri - S3 uri to the workflow template
+  * @param {Object} provider - Cumulus provider object
+  * @param {Object} collection - Cumulus collection object
+  * @param {Object} pdr - the PDR object
+  * @param {string} pdr.name - name of the PDR
```

Comment:
Should we be listing pdr.name if it's not a separate param?

Reply:
this is a jsdoc's feature where you can basically tell people what properties in the pdr object you expect. This helps the developers to to know the pdr object must have a property key called name.